### PR TITLE
added Atomics examples

### PR DESCRIPTION
**Pull request by maboa**
Note I missed a couple out such as wait and wake as it was difficult to create satisfactory examples.

### CODE REVIEW

**Comment by schalkneethling** [comment]
r+ from me. Leaving open for additional review by @wbamberg or @chrisdavidmills

File: live-examples/js-examples/atomics-store.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<pre>
+<code id="static-js" class="language-js">// create a SharedArrayBuffer with a size in bytes
+var buffer = new SharedArrayBuffer(16);
+var uint8 = new Uint8Array(buffer);
+uint8[0] = 5;
+
+console.log(Atomics.store(uint8, 0, 2));
+// expected output: 2
+</code>
```
Comment (wbamberg) [inline live-examples/js-examples/atomics-store.html]: Nit: consider adding a `load` call, to show it's really in there:

```
// create a SharedArrayBuffer with a size in bytes
var buffer = new SharedArrayBuffer(16);
var uint8 = new Uint8Array(buffer);
uint8[0] = 5;

console.log(Atomics.store(uint8, 0, 2));
// expected output: 2

console.log(Atomics.load(uint8, 0));
// expected output: 2
```

File: live-examples/js-examples/atomics-sub.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+<code id="static-js" class="language-js">// create a SharedArrayBuffer with a size in bytes
+var buffer = new SharedArrayBuffer(16);
+var uint8 = new Uint8Array(buffer);
+uint8[0] = 7;
+
+Atomics.sub(uint8, 0, 2); // (returns 7)
+
+console.log(Atomics.load(uint8, 0)); // 7 - 2
+// expected output: 5
+</code>
```
Comment (wbamberg) [inline live-examples/js-examples/atomics-sub.html]: Nit: as for `add` above, consider logging the old value:

```
// create a SharedArrayBuffer with a size in bytes
var buffer = new SharedArrayBuffer(16);
var uint8 = new Uint8Array(buffer);
uint8[0] = 7;

// subtract 2 from the first element, and return the old value
console.log(Atomics.sub(uint8, 0, 2));
// expected output: 7

console.log(Atomics.load(uint8, 0)); // 7 - 2
// expected output: 5
```

File: live-examples/js-examples/atomics-xor.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<pre>
+<code id="static-js" class="language-js">// create a SharedArrayBuffer with a size in bytes
+var buffer = new SharedArrayBuffer(16);
+var uint8 = new Uint8Array(buffer);
+uint8[0] = 7;
+
+Atomics.xor(uint8, 0, 2); // (returns 7)
+
+// 7 (0111) XOR 2 (0010) = 7 (0101)
```
Comment (wbamberg) [inline live-examples/js-examples/atomics-xor.html]: Should be:

`// 7 (0111) XOR 2 (0010) = 5 (0101)`

File: live-examples/js-examples/atomics-and.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<pre>
+<code id="static-js" class="language-js">// create a SharedArrayBuffer with a size in bytes
+var buffer = new SharedArrayBuffer(16);
+var uint8 = new Uint8Array(buffer);
+uint8[0] = 7;
+
+Atomics.and(uint8, 0, 2); // (returns 7)
+
+// 7 (0111) AND 2 (0010) = 0 (0010)
```
[the remark this answers is not in the document]
Comment (maboa) [reply live-examples/js-examples/atomics-and.html]: Good spot. Will correct.

File: live-examples/js-examples/atomics-compareexchange.html [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+var buffer = new SharedArrayBuffer(16);
+var uint8 = new Uint8Array(buffer);
+uint8[0] = 5;
+
+Atomics.compareExchange(uint8, 0, 5, 2); // returns 5
+
+// guarantees the original value by only exchanging if 5
+console.log(Atomics.load(uint8, 0));
+// expected output: 2
+</code>
```
[the remark this answers is not in the document]
Comment (maboa) [reply live-examples/js-examples/atomics-compareexchange.html]: Agree - I think this warrants multiple console.logs. Will change.

File: live-examples/js-examples/atomics-exchange.html [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+<code id="static-js" class="language-js">// create a SharedArrayBuffer with a size in bytes
+var buffer = new SharedArrayBuffer(16);
+var uint8 = new Uint8Array(buffer);
+uint8[0] = 5;
+
+Atomics.exchange(uint8, 0, 2); // returns 5
+
+console.log(Atomics.load(uint8, 0));
+// expected output: 2
+</code>
```
[the remark this answers is not in the document]
Comment (maboa) [reply live-examples/js-examples/atomics-exchange.html]: OK

File: live-examples/js-examples/atomics-islockfree.html [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<pre>
+<code id="static-js" class="language-js">// create a SharedArrayBuffer with a size in bytes
```
[the remark this answers is not in the document]
Comment (maboa) [reply live-examples/js-examples/atomics-islockfree.html]: Oops. Wil fix.

File: live-examples/js-examples/atomics-islockfree.html [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+console.log(Atomics.isLockFree(3));
+// 3 is not one of the BYTES_PER_ELEMENT values
+// expected output: false
+</code>
```
[the remark this answers is not in the document]
Comment (maboa) [reply live-examples/js-examples/atomics-islockfree.html]: Yes. Agreed.

File: live-examples/js-examples/atomics-load.html [hunk 9]
```diff
@@ -0,0 +1,12 @@
+<pre>
+<code id="static-js" class="language-js">// create a SharedArrayBuffer with a size in bytes
+var buffer = new SharedArrayBuffer(16);
+var uint8 = new Uint8Array(buffer);
+uint8[0] = 5;
+
+Atomics.add(uint8, 0, 2); // (returns 5)
+
+console.log(Atomics.load(uint8, 0)); // 5 + 2
```
[the remark this answers is not in the document]
Comment (maboa) [reply live-examples/js-examples/atomics-load.html]: OK.

File: live-examples/js-examples/atomics-store.html (hunk 1 above)
Comment (maboa) [reply live-examples/js-examples/atomics-store.html]: OK

File: live-examples/js-examples/atomics-sub.html (hunk 2 above)
Comment (maboa) [reply live-examples/js-examples/atomics-sub.html]: OK

File: live-examples/js-examples/atomics-xor.html (hunk 3 above)
Comment (maboa) [reply live-examples/js-examples/atomics-xor.html]: Yes. Good spot.

File: live-examples/js-examples/atomics-add.html [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+<code id="static-js" class="language-js">// create a SharedArrayBuffer with a size in bytes
+var buffer = new SharedArrayBuffer(16);
+var uint8 = new Uint8Array(buffer);
+uint8[0] = 7;
+
+Atomics.add(uint8, 0, 2); // (returns 7)
+
+console.log(Atomics.load(uint8, 0)); // 7 + 2
+// expected output: 9
+</code>
```
[the remark this answers is not in the document]
Comment (maboa) [reply live-examples/js-examples/atomics-add.html]: Yes. We can do this now multiple console.logs are an option.

**Comment by wbamberg** [comment]
Thanks for the updates @maboa , this looks great.

**Pull request merged**